app/components/primer/truncate_component.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Primer
+  # The Truncate component is a wrapper component that will shorten text with an ellipsis.
+  class TruncateComponent < Primer::Component
+    # @example 25|Default
+    #   <div class="col-2">
+    #     <%= render(Primer::TruncateComponent.new(tag: :p)) { "branch-name-that-is-really-long" } %>
+    #   </div>
+    #
+    # @example 25|Inline
+    #   <%= render(Primer::TruncateComponent.new(tag: :span, inline: true)) { "branch-name-that-is-really-long" } %>
+    #
+    # @example 25|Expandable
+    #   <%= render(Primer::TruncateComponent.new(tag: :span, inline: true, expandable: true)) { "branch-name-that-is-really-long" } %>
+    #
+    # @param inline [Boolean] Whether the element is inline (or inline-block).
```

Comment:
What do you think about supporting `max_width`, as Primer React does? https://primer.style/components/Truncate

Reply:
I thought about it but wasn't sure if we should have a `max_width` prop or let the user add a `style` tag

Reply:
Hmm. I think it might be nice to declare it as an argument so we can document the use case.

We have some rules that build up a `style` attribute: https://github.com/primer/view_components/blob/main/lib/primer/classify.rb#L200

Reply:
done!